src/vs/workbench/contrib/chat/browser/chatSessions.ts

```diff
@@ -695,7 +697,49 @@ class SessionsViewPane extends ViewPane {
 
 	public refreshTree(): void {
 		if (this.tree && this.isBodyVisible()) {
-			this.tree.updateChildren(this.provider);
+			this.refreshTreeWithProgress();
+		}
+	}
+
+	/**
+	 * Refreshes the tree data with progress indication.
+	 * Shows a progress indicator while the tree updates its children from the provider.
+	 */
+	private async refreshTreeWithProgress(): Promise<void> {
+		if (!this.tree) {
+			return;
+		}
+
+		const progressIndicator = this.getProgressIndicator();
```

Comment:
Not sure this is the right way to do it... maybe should be using `IProgressService`. Unless you see some other code doing it this way